File: docs-examples/csfle-java/src/test/java/io/micronaut/mongodb/docs/NamedClientServiceTest.java
package io.micronaut.mongodb.docs;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.mongodb.testcontainers.MongoDb;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import reactor.core.publisher.Flux;

import java.util.List;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertTrue;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
@MicronautTest(startApplication = false)
class NamedClientServiceTest implements TestPropertyProvider {

    @Override
    public @NonNull Map<String, String> getProperties() {
        return Map.of("mongodb.servers.another.uri", MongoDb.getProperties().get("mongodb.uri"));
    }

    @Inject
    NamedClientService namedClientService;

    @Test
    void listsTheDatabasesOfTheNamedClient() {
        List<String> names = Flux.from(namedClientService.databaseNames()).collectList().block();

        assertTrue(names.contains("admin"), names.toString());
    }
}

File: docs-examples/csfle-kotlin/src/test/kotlin/io/micronaut/mongodb/docs/NamedClientServiceTest.kt
package io.micronaut.mongodb.docs

import io.micronaut.mongodb.testcontainers.MongoDb
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import io.micronaut.test.support.TestPropertyProvider
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance
import reactor.core.publisher.Flux

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
@MicronautTest(startApplication = false)
class NamedClientServiceTest : TestPropertyProvider {

    override fun getProperties(): Map<String, String> =
        mapOf("mongodb.servers.another.uri" to MongoDb.getProperties()["mongodb.uri"]!!)

    @Inject
    lateinit var namedClientService: NamedClientService

    @Test
    fun listsTheDatabasesOfTheNamedClient() {
        val names = Flux.from(namedClientService.databaseNames()).collectList().block()!!

        assertTrue(names.contains("admin"), names.toString())
    }
}
